### PR TITLE
Dev espa v2.15.0

### PR DESCRIPTION
**Pull request by raydittmeier**
pull request/peer review for pixel-based QA water detection tool

### CODE REVIEW

File: cfmask-based-water-detection/src/cfmask_water_detection.c [hunk 1]
```diff
@@ -291,17 +289,24 @@ main (int argc, char *argv[])
         /* If any of the input is fill, make the output fill */
         if (band_red[pixel_index] == red_fill_value ||
             band_nir[pixel_index] == nir_fill_value ||
-            band_class_qa[pixel_index] == class_qa_fill_value)
+            pixel_qa_is_fill(band_pixel_qa[pixel_index]))
```
Comment (gschmidt-usgs) [inline cfmask-based-water-detection/src/cfmask_water_detection.c]: Technically this if-check isn't needed.  If the red or NIR bands are fill, the pixel QA should already be fill.

File: cfmask-based-water-detection/src/const.h [hunk 2]
```diff
@@ -6,21 +6,25 @@
...
+#define L2QA_CLEAR 1
+#define L2QA_WATER 2
+#define L2QA_CLD_SHADOW 3
+#define L2QA_SNOW  4
+#define L2QA_CLOUD 5
+#define L2QA_CLOUD_CONF1 6
+#define L2QA_CLOUD_CONF2 7
+
+#define L2QA_SINGLE_BIT 0x01             /* 00000001 */
+#define L2QA_DOUBLE_BIT 0x03             /* 00000011 */
```
Comment (gschmidt-usgs) [inline cfmask-based-water-detection/src/const.h]: Lines 9-20 are already defined in pixel_qa.h from the Level-2 QA tools.  No need to redefine these.  Just #include the pixel_qa.h file.

File: cfmask-based-water-detection/src/utilities.h [hunk 3]
```diff
@@ -26,6 +29,61 @@
                           __FILE__, __LINE__, stdout); \
             return (status);}
 
+/******************************************************************************
```
Comment (gschmidt-usgs) [inline cfmask-based-water-detection/src/utilities.h]: These new utilities are directly from the level-2 QA tools project.  You could just add that library as a requirement here, but understand if you don't want another requirement for this project.

**Comment by gschmidt-usgs** [comment]
No further comments.

File: cfmask-based-water-detection/src/cfmask_water_detection.c (hunk 1 above)
Comment (raydittmeier) [inline cfmask-based-water-detection/src/cfmask_water_detection.c]: Ok, I'll leave this one alone.

File: cfmask-based-water-detection/src/const.h (hunk 2 above)
Comment (raydittmeier) [inline cfmask-based-water-detection/src/const.h]: Yes, I took these straight from the L2QA code.  I didn't want to make dependencies across projects.

File: cfmask-based-water-detection/src/utilities.h (hunk 3 above)
Comment (raydittmeier) [inline cfmask-based-water-detection/src/utilities.h]: Yes, I took these straight from the L2QA code.  I didn't want to make dependencies across projects.

**Pull request merged**